DataFormats/L1TParticleFlow/interface/PFCandidate.h

```diff
@@ -45,17 +46,42 @@ namespace l1t {
     /// PUPPI weight (-1 if not available)
     float puppiWeight() const { return puppiWeight_; }
 
+    void setZ0(float z0) { setVertex(reco::Particle::Point(0, 0, z0)); }
+    void setDxy(float dxy) { dxy_ = dxy; }
+
+    float z0() const { return vz(); }
+    float dxy() const { return dxy_; }
+
+    int16_t hwZ0() const { return hwZ0_; }
+    int16_t hwDxy() const { return hwDxy_; }
+    uint16_t hwTkQuality() const { return hwTkQuality_; }
+    uint16_t hwPuppiWeight() const { return hwPuppiWeight_; }
+    uint16_t hwEmID() const { return hwEmID_; }
+    uint64_t encodedPuppi64() const { return encodedPuppi64_; }
+
+    void setHwZ0(int16_t hwZ0) { hwZ0_ = hwZ0; }
+    void setHwDxy(int16_t hwDxy) { hwDxy_ = hwDxy; }
+    void setHwTkQuality(uint16_t hwTkQuality) { hwTkQuality_ = hwTkQuality; }
+    void setHwPuppiWeight(uint16_t hwPuppiWeight) { hwPuppiWeight_ = hwPuppiWeight; }
+    void setHwEmID(uint16_t hwEmID) { hwEmID_ = hwEmID; }
+    void setEncodedPuppi64(uint64_t encodedPuppi64) { encodedPuppi64_ = encodedPuppi64; }
+
   private:
     PFClusterRef clusterRef_;
     PFTrackRef trackRef_;
     MuonRef muonRef_;
-    float puppiWeight_;
+    float dxy_, puppiWeight_;
+
+    int16_t hwZ0_, hwDxy_;
+    uint16_t hwTkQuality_, hwPuppiWeight_, hwEmID_;
+    uint64_t encodedPuppi64_;
 
     void setPdgIdFromParticleType(int charge, ParticleType kind);
   };
 
   typedef std::vector<l1t::PFCandidate> PFCandidateCollection;
   typedef edm::Ref<l1t::PFCandidateCollection> PFCandidateRef;
   typedef edm::RefVector<l1t::PFCandidateCollection> PFCandidateRefVector;
+  typedef l1t::RegionalOutput<l1t::PFCandidateCollection> PFCandidateRegionalOutput;
```

Comment:
This definition of `PFCandidateRegionalOutput` is not needed in this class, as far as I can see.
If this line is removed, together with L9 above, then you can move `RegionalOutput` into DataFormats/L1TCorrelator and resolve the circular dependency reported in https://github.com/cms-sw/cmssw/pull/37536/files#r849165513
Is there anything that I miss which prevents it?

Reply:
It is used in [L1Trigger/Phase2L1ParticleFlow/plugins/DeregionizerProducer.cc](https://github.com/cms-sw/cmssw/pull/37536/files#diff-188e0d47aea503e77d5e3a6c5f20cf6e1624a4fe386bf786c35016e8aebbe3fb) https://github.com/cms-sw/cmssw/pull/37536/files#diff-188e0d47aea503e77d5e3a6c5f20cf6e1624a4fe386bf786c35016e8aebbe3fbR38

Reply:
> It is used in [L1Trigger/Phase2L1ParticleFlow/plugins/DeregionizerProducer.cc](https://github.com/cms-sw/cmssw/pull/37536/files#diff-188e0d47aea503e77d5e3a6c5f20cf6e1624a4fe386bf786c35016e8aebbe3fb) https://github.com/cms-sw/cmssw/pull/37536/files#diff-188e0d47aea503e77d5e3a6c5f20cf6e1624a4fe386bf786c35016e8aebbe3fbR38

Exact! Not in PFCandidate, then.
Please move the dependency over there